modules/Oculus_module/src/RobotControlHelper.cpp

```diff
@@ -0,0 +1,300 @@
+        remoteControlBoardsList.addString("/" + robot + "/" + iCubPart);
+
+    options.put("remoteControlBoards", remoteControlBoards.get(0));
+    options.put("localPortPrefix", "/" + name + "/remoteControlBoard");
+    yarp::os::Property& remoteControlBoardsOpts = options.addGroup("REMOTE_CONTROLBOARD_OPTIONS");
+    remoteControlBoardsOpts.put("writeStrict", "on");
+
+    m_actuatedDOFs = axesList.size();
+
+    // open the device
+    if (!m_robotDevice.open(options))
+    {
+        yError() << "[RobotControlHelper::configure] Could not open remotecontrolboardremapper "
+                    "object.";
+        m_isActive = false;
+        return false;
+    }
+
+    if (!m_robotDevice.view(m_encodersInterface) || !m_encodersInterface)
+    {
+        yError() << "[RobotControlHelper::configure] Cannot obtain IEncoders interface";
+        m_isActive = false;
+        return false;
+    }
+
+    if (!m_robotDevice.view(m_positionInterface) || !m_positionInterface)
+    {
+        yError() << "[RobotControlHelper::configure] Cannot obtain IPositionControl interface";
+        m_isActive = false;
+        return false;
+    }
+
+    if (!m_robotDevice.view(m_positionDirectInterface) || !m_positionDirectInterface)
+    {
+        yError() << "[RobotControlHelper::configure] Cannot obtain IPositionDirect interface";
+        m_isActive = false;
+        return false;
+    }
+
+    if (!m_robotDevice.view(m_limitsInterface) || !m_limitsInterface)
+    {
+        yError() << "[RobotControlHelper::configure] Cannot obtain IPositionDirect interface";
+        m_isActive = false;
+        return false;
+    }
+
+    if (!m_robotDevice.view(m_controlModeInterface) || !m_controlModeInterface)
+    {
+        yError() << "[RobotControlHelper::configure] Cannot obtain IControlMode interface";
+        m_isActive = false;
+        return false;
+    }
+
+    if (!m_robotDevice.view(m_timedInterface) || !m_timedInterface)
+    {
+        yError() << "[RobotControlHelper::configure] Cannot obtain iTimed interface";
+        m_isActive = false;
+        return false;
+    }
+
+    m_desiredPositionInDegrees.resize(m_actuatedDOFs);
+    m_positionFeedbackInDegrees.resize(m_actuatedDOFs);
+    m_positionFeedbackInRadians.resize(m_actuatedDOFs);
+    m_minJointsPosition.resize(m_actuatedDOFs);
+    m_maxJointsPosition.resize(m_actuatedDOFs);
+
+    // check if the robot is alive
+    bool okPosition = false;
+    for (int i = 0; i < 10 && !okPosition; i++)
+    {
+        okPosition = m_encodersInterface->getEncoders(m_positionFeedbackInDegrees.data());
+
+        if (!okPosition)
+            yarp::os::Time::delay(0.1);
+    }
+    if (!okPosition)
+    {
+        yError() << "[configure] Unable to read encoders (position).";
+        m_isActive = false;
+        return false;
+    }
+
+    for (int i = 0; i < m_actuatedDOFs; i++)
+        // get position limits
+        if (!m_limitsInterface->getLimits(i, &m_minJointsPosition(i), &m_maxJointsPosition(i)))
```

Comment:
I would get the limits in the ``getLimits`` method. In this way the ``configure`` phase does not fail if some joints are in the "not configured" status. Also, we can save a vector of bools to understand which joint is currently not working.

Reply:
Yes, I can move the `getLimts()` inside the `RobotControlHelper::getLimits()`. Indeed the limits are not used by the `RobotControlHelper` class itself (except [here](https://github.com/GiulioRomualdi/walking-teleoperation/blob/refactoring/oculus_module/modules/Oculus_module/src/RobotControlHelper.cpp#L197-L201) but as discussed above we may remove those lines)

Reply:
Done in 185a6dd